Update sync docs (#214)
tidalmigrations/tidalmigrations.github.io: pages/tidal tools/syncservers.md

folder: tidaltools
---

## What is Syncing?

> Syncing is a process that creates and updates your inventories to your Tidal Migrations Workspace. This is useful if you

Reviewer: ```suggestion
Syncing is a process that creates and updates your inventories in your Tidal Migrations Workspace. This is useful if you
```

... I think?

would like to update a lot of data across many different records. It can also be useful to keep your inventory up to date with an external

> date source, by continually updating your inventory on a recurring scheudle.

Reviewer: ```suggestion
date source, by continually updating your inventory on a recurring schedule.
```

## How do I sync my servers?

> {% include note.html content="You will first need to have [Tidal Tools installed and login](https://guides.tidalmg.com/tidal-tools.html) to your Workspace." %}

Reviewer: ```suggestion
{% include note.html content="You will first need to have [Tidal Tools installed and logged in](https://guides.tidalmg.com/tidal-tools.html) to your Workspace." %}
```

You can sync any data source with Tidal Migrations by generating a simple JSON document of the data.

This data can be passed as standard input to the `tidal sync servers` command and your servers data will be
synchronized via the API.

The JSON document must be created in the specified format. The top-level key must be `"servers"`,
with a value of an array. The array can consist of the various keys as shown below, describing the server to be synced.
You can also include any other arbitrary fields in the key `"custom_fields"`.

{% include note.html content="If you pass data to Tidal Tools via standard input (ie. `cat x.json | tidal sync servers`)

> any custom fields presented will be automatically created if they don't already exist. If you provide the data to the

Reviewer: I'll have to take your word on this, most of our experience in this area was MS accuracy / limitation tests, we didn't get too close to the custom field creation logic. I remember there was some confusion around the docs on this last week right @jmariomejiap?

Reviewer 2: Does the CLI do this? I don't remember the backend creating new fields.

tidal sync servers some_file.json
```

> Any servers provided that already exist, based on their hostname, will be updated with the data

Reviewer: Nice 👍

> provided in the JSON document. If no server with the given hostname exist, that server will be created with all the included data.

Reviewer: ```suggestion
provided in the JSON document. If no server with the given hostname exists, that server will be created with all the included data.
```
